### PR TITLE
Don't set a fake parent when parsing `{% embed "foo.html.twig." %}`

### PR DESCRIPTION
**Pull request by wimleers**
Context: #2059.

### CODE REVIEW

File: lib/Twig/TokenParser/Embed.php [hunk 1]
```diff
@@ -22,19 +22,15 @@ public function parse(Twig_Token $token)
 
         list($variables, $only, $ignoreMissing) = $this->parseArguments();
 
-        // inject a fake parent to make the parent() function work
         $stream->injectTokens(array(
             new Twig_Token(Twig_Token::BLOCK_START_TYPE, '', $token->getLine()),
             new Twig_Token(Twig_Token::NAME_TYPE, 'extends', $token->getLine()),
-            new Twig_Token(Twig_Token::STRING_TYPE, '__parent__', $token->getLine()),
+            new Twig_Token(Twig_Token::STRING_TYPE, $parent->getAttribute('value'), $token->getLine()),
```
Comment (stof) [inline lib/Twig/TokenParser/Embed.php]: this is broken, because it works only for a ConstantNode, while we support any expression

**Comment by stof** [comment]
the fact that tests are passing tells me that we are missing some tests here (similar to the ones we have for ``extends``)

**Comment by wimleers** [comment]
+1

**Comment by fabpot** [comment]
@wimleers I can try to work on a fix tonight if you cannot find a way to fix it.

**Comment by wimleers** [comment]
@fabpot Feel free to do so — I don't know the Twig codebase well enough.

**Comment by fabpot** [comment]
In #2124, I've made something that works for when the parent is indeed a constant.

**Pull request closed without merging**